controllers/auth.js

```diff
@@ -0,0 +1,63 @@
+const bcrypt = require('bcryptjs');
+const nodemailer = require('nodemailer');
+const sendGridTransport = require('nodemailer-sendgrid-transport');
+const { validationResult } = require('express-validator');
+
+const User = require('../models/user');
+
+const transporter = nodemailer.createTransport(
+  sendGridTransport({
+    auth: {
+      api_key: process.env.SEND_GRID_API_KEY
+    }
+  })
+);
+
+exports.signup = (req, res, next) => {
+  const { email, password } = req.body;
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res.status(409).send(errors.array()[0].msg);
+  }
+
+  return bcrypt
```

Comment:
better to use async await for this whole controller. It will reduce the code complexity  a lot.